### DISCUSSION TITLE
airflow upgrade from 1.10.15 to 2.2.3 failed

### DISCUSSION THREAD

**Discussion by GreeMapengfei** [apache/airflow#22893] [Q&A]
problem:
I am currently using airflow1.10 to upgrade to 2.23, but I am reporting this error when running the upgrade plug-in
error:

when i run airflow upgrade_check can meet
AttributeError: 'AppBuilderCodeEditorView' object has no attribute 'datamodel'

this is update plugin caused i guess

**Comment by potiuk** [comment]
This is your plugin's problem. If the plugin used some internal Airflow code, it needs to be updated to match the new version

**Reply by potiuk** [reply]
I mean that somebody in your team wrote a custom code that is not compatible. Whoever wrote it, has to fix it.

**Comment by eladkal** [comment]
During the upgrade it was buggy for me as well.
I simply excluded this specific test:
`airflow upgrade_check --ignore UseCustomSQLAInterfaceClassRule`
The upgrade check is just an assistant tool, so I also double checked everything to make sure upgrade works.
